packages/react-button/src/components/Button/useButtonClasses.ts

```diff
@@ -1,489 +1,403 @@
+        fontSize: buttonTokens.base.fontSize,
+        lineHeight: buttonTokens.base.lineHeight,
+      };
+    },
+  ],
+  [
+    ({ size }) => size === 'small',
+    theme => {
+      const buttonTokens = makeButtonTokens(theme);
+
+      return {
+        fontSize: buttonTokens.small.fontSize,
+        fontWeight: buttonTokens.small.fontWeight,
+        lineHeight: buttonTokens.small.lineHeight,
+      };
+    },
+  ],
+  [
+    ({ size }) => size === 'large',
+    theme => {
+      const buttonTokens = makeButtonTokens(theme);
+
+      return {
+        fontSize: buttonTokens.large.fontSize,
+        lineHeight: buttonTokens.large.lineHeight,
+      };
+    },
+  ],
+]);
 
-        pressed: {
-          background: palette?.neutralLight,
-          contentColor: palette?.black,
-          iconColor: palette?.themeDarker,
-        },
+const useIconClasses = makeStyles<ButtonStyleSelectors>([
+  [
+    null,
+    theme => {
+      const buttonTokens = makeButtonTokens(theme);
 
-        highContrast: {
-          background: 'Window',
-          borderColor: 'transparent',
-          contentColor: 'WindowText',
-          iconColor: 'WindowText',
-
-          hovered: {
-            background: 'Window',
-            borderColor: 'transparent',
-            contentColor: 'Highlight',
-            iconColor: 'Highlight',
-          },
-
-          pressed: {
-            background: 'Window',
-            borderColor: 'transparent',
-            contentColor: 'Highlight',
-            iconColor: 'Highlight',
-          },
-
-          disabled: {
-            background: 'Window',
-            borderColor: 'transparent',
-            contentColor: 'GrayText',
-            iconColor: 'GrayText',
-          },
-        },
-      },
-
-      // TODO: no references to palette.
-      transparent: {
-        background: 'transparent',
-        borderColor: 'transparent',
-        contentColor: palette?.neutralPrimary,
-        fontWeight: 'normal',
-        iconColor: palette?.themeDarkAlt,
-
-        disabled: {
-          background: 'transparent',
-          borderColor: 'transparent',
-          contentColor: palette.neutralTertiary,
-          iconColor: 'inherit',
-        },
+      return {
+        display: 'inline-flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        height: buttonTokens.base.iconHeight,
+        width: buttonTokens.base.iconWidth,
+      };
+    },
+  ],
+  [
+    ({ size }) => size === 'large',
+    theme => {
+      const buttonTokens = makeButtonTokens(theme);
+
+      return {
+        width: buttonTokens.large.iconWidth,
+        height: buttonTokens.large.iconHeight,
+      };
+    },
+  ],
+]);
 
-        hovered: {
-          background: 'transparent',
-          contentColor: palette?.themePrimary,
-          iconColor: palette?.themePrimary,
-        },
+export const useButtonClasses = (state: ButtonState, selectors: ButtonStyleSelectors) => {
```

Comment:
```suggestion
export const useButtonStyles = (state: ButtonState, selectors: ButtonStyleSelectors) => {
```

Can you please rename it to have `*Styles` suffix? It's a pattern followed in other components. And file, too 🐱

Reply:
Done, thanks